src/Apps/Order/Routes/Shipping2/ShippingContext.ts

```diff
@@ -1,20 +1,342 @@
+  // are less relevant if the user has saved addresses - Setting country
+  // doesn't matter.
+  const initialFulfillmentValues: ShipValues["attributes"] = {
+    ...addressWithFallbackValues({ country: orderData.shipsFrom }),
+
+    addressVerifiedBy: null,
+    saveAddress: savedAddresses.length === 0,
+  }
+
+  return {
+    shippingQuotes: initialShippingQuotes,
+    fulfillmentDetails: {
+      fulfillmentType: FulfillmentType.SHIP,
+      attributes: initialFulfillmentValues,
+    },
+  }
+}
+
+const matchAddressFields = (...addressPair: [object, object]) => {
+  const [a1, a2] = addressPair.map(a => addressWithFallbackValues(a))
+  return (
+    a1.addressLine1 === a2.addressLine1 &&
+    a1.addressLine2 === a2.addressLine2 &&
+    a1.city === a2.city &&
+    a1.country === a2.country &&
+    a1.name === a2.name &&
+    a1.phoneNumber === a2.phoneNumber &&
+    a1.postalCode === a2.postalCode &&
+    a1.region === a2.region
+  )
+}
+
+const getSavedFulfillmentData = (
+  order: ShippingProps["order"],
+  me: ShippingProps["me"]
+): {
+  fulfillmentType: FulfillmentType
+  isArtsyShipping: boolean
+  fulfillmentDetails: FulfillmentValues["attributes"]
+  selectedSavedAddressId: string | null
+} | null => {
+  if (
+    !order.requestedFulfillment ||
+    Object.keys(order.requestedFulfillment).length === 0
+  ) {
+    return null
+  }
+
+  const requestedFulfillmentType = order.requestedFulfillment.__typename
+  if (requestedFulfillmentType === "CommercePickup") {
+    const phoneNumber = order.requestedFulfillment.phoneNumber!
+    return {
+      fulfillmentType: FulfillmentType.PICKUP,
+      isArtsyShipping: false,
+      // TODO: [When things are working again]
+      // figure out what `name` is used for w/ pickup, where to get it from
+      fulfillmentDetails: { phoneNumber } as FulfillmentValues["attributes"],
+      selectedSavedAddressId: null,
+    }
+  }
+  const fulfillmentDetails: ShippingAddressFormValues = addressWithFallbackValues(
+    order.requestedFulfillment
+  )
+
+  const addressList =
+    extractNodes(me?.addressConnection) ??
+    ([] as SavedAddressType[]).filter(a => !!a)
+
+  // we don't store the address id on exchange orders, so we need to match every field
+  const selectedSavedAddressId =
+    addressList.find(node => matchAddressFields(node, fulfillmentDetails))
+      ?.internalID || null
+
+  if (requestedFulfillmentType === "CommerceShipArta") {
+    return {
+      fulfillmentType: FulfillmentType.SHIP,
+      isArtsyShipping: true,
+      fulfillmentDetails,
+      selectedSavedAddressId,
+    }
+  }
+  if (requestedFulfillmentType === "CommerceShip") {
+    return {
+      fulfillmentType: FulfillmentType.SHIP,
+      isArtsyShipping: false,
+      fulfillmentDetails,
+      selectedSavedAddressId,
+    }
+  }
+  // Should never happen. Log it?
+  return null
+}
+
+// Compute and memoize data from the saved order.
+export const useLoadComputedData = (
```

Comment:
We should remove `load` from the name here as it is not actually fetching anything